frontend/src/pages/CouponList/index.tsx

```diff
@@ -52,11 +59,34 @@ const CouponList = () => {
   const [currentIndex, setCurrentIndex] = useState(0);
   const couponListContainerRef = useRef<HTMLDivElement>(null);
   const [isLast, setIsLast] = useState(false);
-  const { data, status } = useQuery<{ coupons: CouponType[] }>(['coupons'], getCoupons, {});
+  const [alertMessage, setAlertMessage] = useState('');
+  const queryClient = useQueryClient();
+  const { data, status } = useQuery<{ coupons: CouponType[] }>(['coupons'], getCoupons, {
+    onSuccess: (data) => {
+      setCurrentIndex(data.coupons.length - 1);
+    },
+  });
+  const { mutate: mutateIsFavorites } = useMutation(
+    ({ cafeId, isFavorites }: PostIsFavoritesReq) => postIsFavorites({ cafeId, isFavorites }),
+    {
+      onSuccess: () => {
+        closeModal();
+      },
+      onMutate: async () => {
+        await queryClient.cancelQueries(['coupons']);
+        queryClient.setQueryData(['coupons'], ({ coupons }: any) => {
+          coupons[currentIndex].couponInfos[0].isFavorites =
+            !coupons[currentIndex].couponInfos[0].isFavorites;
+        });
+      },
+    },
+  );
```

Comment:
setQueryData의 Updater 부분이 반환하는 값에 따라 어떨지 몰라서 일단 주석으로 해놨습니다.
타입을 찾아보면
```ts
setQueryData<TQueryFnData>(
    queryKey: QueryKey,
    updater: Updater<TQueryFnData | undefined, TQueryFnData | undefined>,
    options?: SetDataOptions,
  ): TQueryFnData | undefined

export type Updater<TInput, TOutput> =
  | TOutput
  | DataUpdateFunction<TInput, TOutput>

export type DataUpdateFunction<TInput, TOutput> = (input: TInput) => TOutput
```
이기 때문에 다음과 같이 고칠수 있을 것 같습니다
```suggestion
queryClient.setQueryData<{ coupons: CouponType[] }>(['coupons'], (param) => {
          if (!param) return;
          param.coupons[currentIndex].couponInfos[0].isFavorites =
            !param.coupons[currentIndex].couponInfos[0].isFavorites;
          return undefined; // return undefined or { coupons: CouponType[] }
        });
```

Reply:
역시 타스 선생.. 윤생..